src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -248,7 +248,10 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
                         type = type.WithTypeAndModifiers(
                             CustomModifierUtils.CopyTypeCustomModifiers(overriddenPropertyType.Type, type.Type, this.ContainingAssembly),
                             overriddenPropertyType.CustomModifiers);
-                        _lazyType = default;
+
+                        // Although we only do this in error scenarios, it is undesirable to mutate the symbol by setting its type twice.
+                        // Tracked by https://github.com/dotnet/roslyn/issues/35381
+                        _lazyType.InterlockedReset();
```

Comment:
>_lazyType.InterlockedReset(); [](start = 24, length = 29)

FYI I think we could keep doing `_lazyType = default;` here.

Also, I don't think this setting of value is the cause of the observed corefx bug, because this only affects an error OHI case.

Reply:
I agree with that.